src/main/java/core/basesyntax/servise/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.servise;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String REPORT_HEADER = "fruit,quantity" + System.lineSeparator();
+    private final Map<String, Integer> storage;
```

Comment:
The `storage` field should be private to ensure encapsulation. If other classes need to access this data, consider providing a method that returns a read-only view of the storage.